File: packages/logic-python/word_of_day.py
import json
from datetime import datetime
from pathlib import Path

_word_list_cache = None


def get_day_number(reference: datetime = datetime(2025, 1, 1)) -> int:
    today = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
    ref = reference.replace(hour=0, minute=0, second=0, microsecond=0)
    return (today - ref).days


def load_word_list() -> list[dict]:
    global _word_list_cache
    if _word_list_cache is not None:
        return _word_list_cache
    data_path = Path(__file__).parent.parent.parent / "data" / "word_list.json"
    with open(data_path, encoding="utf-8") as f:
        _word_list_cache = json.load(f)
    return _word_list_cache


def get_word_of_the_day() -> dict:
    word_list = load_word_list()
    day_number = get_day_number()
    index = day_number % len(word_list)
    return word_list[index]

File: packages/logic-next/src/lib/game/word-of-day.ts
import path from 'path'
import fs from 'fs'
import type { WordData } from '@shared/types/word-data'

// new Date(2025, 0, 1) is local midnight, matching Python's datetime(2025, 1, 1)
function getDayNumber(referenceDate: Date = new Date(2025, 0, 1)): number {
  const today = new Date()
  today.setHours(0, 0, 0, 0)
  referenceDate.setHours(0, 0, 0, 0)
  const diffTime = today.getTime() - referenceDate.getTime()
  const diffDays = Math.floor(diffTime / (1000 * 60 * 60 * 24))
  return diffDays
}

let wordListCache: WordData[] | null = null

function loadWordList(): WordData[] {
  if (wordListCache) return wordListCache
  const dataPath = path.join(process.cwd(), '..', '..', 'data', 'word_list.json')
  const data = fs.readFileSync(dataPath, 'utf-8')
  wordListCache = JSON.parse(data)
  return wordListCache!
}

export function getWordOfTheDay(): WordData {
  const wordList = loadWordList()
  const dayNumber = getDayNumber()
  const index = dayNumber % wordList.length
  return wordList[index]
}
